pandas/core/groupby/generic.py

```diff
@@ -108,7 +108,12 @@
 ScalarResult = TypeVar("ScalarResult")
 
 
-class NamedAgg(NamedTuple):
+class _BaseNamedAgg(NamedTuple):
```

Comment:
What's the reason for introducing a new class here?

Reply:
That's because you cannot override __new__ when inheriting from NamedTuple directly.

Reply:
So the only way I found to override it while maintaining the NamedTuple properties was to Introduce a new class. Do you have a suggestion for another method to do it?

Reply:
the reference error for when I tried to inherit directly from NamedTuple is:
`AttributeError: Cannot overwrite NamedTuple attribute __new__`

Reply:
Ah that's unfortunate. This seems kind of hacky as a result though - maybe we can just drop the NamedTuple subclass though? I think that was mostly as a convenience rather than an intentional design decision.

@rhshadrach might have more thoughts

Reply:
Apply does expect a tuple, so to remove this behavior we would need to make abigger change.

Reply:
https://github.com/python/cpython/issues/77258 has some relevant information, especially the linked discussion there to the issue in `python/typing`.

In any case, it looks to me that this implementation just ignores `args` / `kwargs` when the provided `aggfunc` is not a callable? If that's the case, then I'm -1 on this approach anyways.

Reply:
@rhshadrach I'll rethink the approach to include the function names passed as string.

I'm not sure I understand your thoughts on the necessity of keeping NamedArgs with tuple behavior. Could you clarify that? 

TY

Reply:
> I'm not sure I understand your thoughts on the necessity of keeping NamedArgs with tuple behavior. Could you clarify that?

I don't believe I shared any thoughts in this regard, just a link to some background information on why Python raises when you try to overwrite `__new__`.